packages/common/upgrade/src/params.ts

```diff
@@ -198,7 +198,8 @@ export class AngularJSUrlCodec implements UrlCodec {
   // https://github.com/angular/angular.js/blob/864c7f0/src/ng/urlUtils.js#L60
   parse(url: string, base?: string) {
     try {
-      const parsed = new URL(url, base);
+      // Safari 12 throws an error when the URL constructor is called with an undefined base.
+      const parsed = (base === undefined) ? new URL(url) : new URL(url, base);
```

Comment:
I note on MDN that the `base` argument on the `URL` constructor defaults to `''` if not provided (https://developer.mozilla.org/en-US/docs/Web/API/URL/URL#Parameters).

So how about we change the fix to be

```suggestion
      const parsed = new URL(url, base || '');
```